src/coreclr/tools/Common/TypeSystem/Common/PortableExecutableMethodExtensions.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Reflection.PortableExecutable;
+
+namespace Internal.TypeSystem
+{
+    public static class PortableExecutableMethodExtensions
+    {
+        public static ImmutableArray<DebugDirectoryEntry> SafeReadDebugDirectory(this PEReader peReader)
+        {
+            int actualDbgDirSize = peReader.PEHeaders.PEHeader.DebugTableDirectory.Size;
+
+            // This comes from the Size property of the DebugDirectoryEntry class.
+            int expectedDbgDirSizeBase = 28;
```

Comment:
Nit: Might use `const` here:
```suggestion
            const int ExpectedDbgDirSize = 28;
```